src/NServiceBus.Core/Persistence/InMemory/Gateway/InMemoryGatewayDeduplicationFeature.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace NServiceBus.Features
+{
+    using Gateway.Deduplication;
+
+    /// <summary>
+    /// Used to configure in memory outbox persistence.
+    /// </summary>
+    public class InMemoryGatewayDeduplicationFeature : Feature
+    {
+        internal InMemoryGatewayDeduplicationFeature()
+        {
+            //DependsOn<gateway>();
+            Defaults(s => s.EnableFeature(typeof(InMemoryGatewayDeduplicationFeature)));
+        }
+
+        /// <summary>
+        /// See <see cref="Feature.Setup" />.
+        /// </summary>
+        protected internal override void Setup(FeatureConfigurationContext context)
+        {
+            if(!context.Settings.TryGet<int>(MaxSizeKey,out var maxSize))
```

Comment:
The default max side can be registered as a default value so we don't need to do a check here.